lib/rubocop/cop/internal_affairs/undefined_config.rb

```diff
@@ -8,7 +8,7 @@ class UndefinedConfig < Base
         ALLOWED_CONFIGURATIONS = %w[
           Safe SafeAutoCorrect AutoCorrect Severity StyleGuide Details Reference Include Exclude
         ].freeze
-        RESTRICT_ON_SEND = %i[[] fetch].freeze
+        RESTRICT_ON_SEND = [:[], :fetch].freeze
```

Comment:
If bracket do not contain spaces, they should not be changed.

So, only the following case that need escaping can be changed from percent literal style to bracket style:

```ruby
%i[[ ]] #=> [:"[", :"]"]
```

This case where brackets do not contain spaces should not be changed:

```ruby
%i[[]]  #=> [:[]]
```

That is, it must be checked for spaces in brackets. Can you update the entire changes in this PR?

Reply:
@koic I think that makes sense to me!

But one thing that I notice is that while `%i[[] one]` does not currently raise an offense, rubocop still autocorrects `[:[], :one]` to `%i[\[\] one]`, which feels inconsistent 🤔 Any thoughts?

Reply:
When `EnforcedStyle: percent`, `%i[[] one]` (no spaces in the brackets) will need to adjust the implementation if necessary to be accepted.